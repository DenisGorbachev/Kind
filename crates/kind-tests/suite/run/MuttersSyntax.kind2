type Sum {
    a
    b
    c
}

#derive[mutters]
record Identity (t: Type) {
    value: t
}

#derive[mutters]
record NoTypeVar {
    some_thing: Identity (Identity U60)
}

Main : NoTypeVar
Main =
    let f = NoTypeVar.new (Identity.new (Identity.new 100))
    !NoTypeVar f .some_thing .value .value @= x => (+ x 200)