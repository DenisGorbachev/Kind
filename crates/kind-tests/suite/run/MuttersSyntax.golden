(NoTypeVar.new (Identity.new (Identity.new 300)))